crate_universe/private/util.bzl

```
@@ -79,23 +86,58 @@ def get_host_info(repository_ctx):
         "windows": ["x86_64"],
     }
 
-    # The expected file extension of crate resolver binaries
-    extension = ""
-
     if "linux" in repository_ctx.os.name:
         cpu = _query_cpu_architecture(repository_ctx, supported_architectures["linux"])
+        host_triple = "{}-unknown-linux-gnu".format(cpu)
         resolver_triple = "{}-unknown-linux-gnu".format(cpu)
-        toolchain_repo = "@rust_linux_{}".format(cpu)
     elif "mac" in repository_ctx.os.name:
         cpu = _query_cpu_architecture(repository_ctx, supported_architectures["macos"])
+        host_triple = "{}-apple-darwin".format(cpu)
         resolver_triple = "{}-apple-darwin".format(cpu)
-        toolchain_repo = "@rust_darwin_{}".format(cpu)
     elif "win" in repository_ctx.os.name:
         cpu = _query_cpu_architecture(repository_ctx, supported_architectures["windows"], True)
+
+        # TODO: The resolver triple should be the same as the host but for the time being,
+        # the resolver is compiled with `-gnu` not `-msvc`.
+        host_triple = "{}-pc-windows-msvc".format(cpu)
         resolver_triple = "{}-pc-windows-gnu".format(cpu)
-        toolchain_repo = "@rust_windows_{}".format(cpu)
-        extension = ".exe"
     else:
         fail("Could not locate resolver for OS " + repository_ctx.os.name)
 
-    return (resolver_triple, toolchain_repo, extension)
+    return (host_triple, resolver_triple)
+
+def get_cargo_and_rustc(repository_ctx, host_triple):
+    """Download a cargo and rustc binary based on the host triple.
```

```suggestion
    """Retrieve a cargo and rustc binary based on the host triple.
```

or locate, maybe... Downloading implies this always causes a download, which I don't think is the case?